mlrun/datastore/targets.py

```diff
@@ -341,8 +343,18 @@ def _get_column_list(self, features, timestamp_key, key_columns, with_type=False
         return result
 
     def write_dataframe(
-        self, df, key_column=None, timestamp_key=None, **kwargs,
+        self, df, key_column=None, timestamp_key=None, chunk_id=0, **kwargs,
     ) -> typing.Optional[int]:
+        def path_with_chunk():
```

Comment:
This function is already very long, I don't see a reason not moving this to a separate private method